fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/rules/ImplementInJoinRule.java

```diff
@@ -0,0 +1,354 @@
+                    newInnerPlanReference = GroupExpressionRef.of(inJoinPlan);
+                }
+
+                call.yield(newInnerPlanReference);
+            }
+        }
+    }
+
+    @Nonnull
+    @SuppressWarnings("unchecked")
+    private ImmutableList<InSource> getInSourcesForRequestedOrdering(@Nonnull final Map<CorrelationIdentifier, Quantifier> explodeAliasToQuantifierMap,
+                                                                     @Nonnull final Set<CorrelationIdentifier> explodeAliases,
+                                                                     @Nonnull final IdentityBiMap<Quantifier.ForEach, ExplodeExpression> quantifierToExplodeBiMap,
+                                                                     @Nonnull final Ordering providedInnerOrdering,
+                                                                     @Nonnull final RequestedOrdering requestedOrdering) {
+        final var availableExplodeAliases = Sets.newLinkedHashSet(explodeAliases);
+
+        final var requestedOrderingKeyParts = requestedOrdering.getOrderingKeyParts();
+        final var sourcesBuilder = ImmutableList.<InSource>builder();
+        final var resultOrderingKeyPartsBuilder = ImmutableList.<KeyPart>builder();
+        final var innerOrderingKeyParts = providedInnerOrdering.getOrderingKeyParts();
+        final var innerEqualityBoundKeyMap = providedInnerOrdering.getEqualityBoundKeyMap();
+        final var resultOrderingEqualityBoundKeyMap  =
+                HashMultimap.create(innerEqualityBoundKeyMap);
+
+        for (var i = 0; i < requestedOrderingKeyParts.size() && !availableExplodeAliases.isEmpty(); i++) {
+            final var requestedOrderingKeyPart = requestedOrderingKeyParts.get(i);
+            final var comparisons = innerEqualityBoundKeyMap.get(requestedOrderingKeyPart.getNormalizedKeyExpression());
+            if (comparisons.isEmpty()) {
+                return ImmutableList.of();
+            }
+
+            final var comparisonsCorrelatedTo = comparisons.stream()
+                    .flatMap(comparison -> comparison.getCorrelatedTo().stream())
+                    .collect(ImmutableSet.toImmutableSet());
+
+            if (comparisonsCorrelatedTo.size() > 1) {
+                return ImmutableList.of();
+            }
+
+            if (Sets.intersection(comparisonsCorrelatedTo, explodeAliases).isEmpty()) {
+                //
+                // This case covers comparisons such as  a = <literal> or a = <correlation> where
+                // <correlation> is anchored above and therefore this comparison causes an equality-bound
+                // expression.
+                //
+                // Example: requested ordering: a, b, c
+                //          ins: a explodes over (1, 2)
+                //               c explodes over ('x', 'y')
+                //          plan has equality-bound expressions over a, b, c
```

Comment:
Why is `b` bound?

Reply:
Elaborated a little bit:

```
//
// This case covers comparisons such as  a = <literal> or a = <correlation> where
// <correlation> is anchored above and therefore this comparison causes an equality-bound
// expression.
//
// Example: requested ordering: a, b, c
//          INs: a explodes over (1, 2)
//               c explodes over ('x', 'y')
//          inner plan has equality-bound expressions over a, b, c
//
// b is not among the INs, but it is bound via equality somehow. That binding can be a constant or
// another correlation that is anchored above the current SELECT.
//
```

The plan at consideration as equality-bound fields `a, b, c`. That is given. If we look at `b`, `b` just looks like it is a comparison to something but nothing related to the INs in this expression. We should just ignore those fields. (That also is true for things like `b = 5`.